### DISCUSSION TITLE
"GPU Selection Issue: NVIDIA Card Preferred Over More Powerful AMD Card"[Bug]:

### DISCUSSION THREAD

**Discussion by 10100programer** [lshqqytiger/stable-diffusion-webui-directml#238] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits

### What happened?

I've encountered a rather unusual scenario.

My setup includes an AMD 7950XTX and a NVIDIA 1080ti. Strangely, the system tends to favor the NVIDIA card despite the AMD card being significantly more powerful.

I've searched extensively for guidance on how to manually designate the preferred card, but unfortunately, I haven't come across any relevant information. I've categorized this as a bug report, although I'm sorry if this classification is not accurate.

### Steps to reproduce the problem

1. Start stable-diffusion-webui-directml  using webui-user.bat
2. Type "steampunk" into prompt
3. Press Generate
4. Observe Memory and Temp of NVIDIA 1080ti go up
![image](https://github.com/lshqqytiger/stable-diffusion-webui-directml/assets/23159784/4931ef9d-b777-4e06-ae9d-85aceed1b5e0)

### What should have happened?

I might have anticipated it to either perform a rapid hash test to determine the optimal card for usage, or to inquire the user about which card to utilize, or even to employ both cards.

### Version or Commit where the problem happens

58b9fb5

### What Python version are you running on ?

Python 3.10.x

### What platforms do you use to access the UI ?

Windows

### What device are you running WebUI on?

NVIDIA GPUs, AMD GPUs

### Cross attention optimization

Automatic

### What browsers do you use to access the UI ?

Mozilla Firefox

### Command Line Arguments

```Shell
--autolaunch
```

### List of extensions

none

### Console logs

```Shell
N/A
```

**Comment by lshqqytiger** [comment]
Add `--backend directml --device-id 1`.
`--backend directml`: Prevent CUDA detection.
`--device-id 1`: Explicitly select RX 7900 XTX because GTX 1080 Ti can work with DirectML.
